components/SpotInfo/index.js

```diff
@@ -15,32 +16,31 @@ import {
   EditIcon,
   NoEntryMessage,
   SpotDeleteButton,
-  SpotNameChangedSuccess,
+  ModalHeadline,
+  ModalMessage,
+  ModalDeleteButton,
+  ModalKeepButton,
 } from "./style";
-import LoadingSpinner from "../LoadingSpinner";
 import Error from "../Error";
 import { CiEdit, CiCircleCheck } from "react-icons/ci";
 import EditDeleteInfoForm from "../EditDeleteInfoForm";
 import AddNewInfoForm from "../AddNewInfoForm";
+import Modal from "../Modal";
 
 export default function SpotInfo({ spotId }) {
-  const {
-    data: spot,
-    error,
-    isValidating,
-    mutate,
-  } = useSWR(`/api/spots/${spotId}`);
+  const { data: spot, error, mutate } = useSWR(`/api/spots/${spotId}`);
   const [isEditingSpotName, setIsEditingSpotName] = useState(false);
   const [spotNameError, setSpotNameError] = useState("");
   const [newSpotName, setNewSpotName] = useState("");
   const [spotNameChangeSuccess, setSpotNameChangeSuccess] = useState(false);
   const router = useRouter();
+  const [isModalOpen, setIsModalOpen] = useState(false);
```

Comment:
Very minor bit of house keeping, but i would group all your useStates together. i.e. 

``` js
const router = useRouter();

const [isEditingSpotName, setIsEditingSpotName] = useState(false);
const [spotNameError, setSpotNameError] = useState("");
const [newSpotName, setNewSpotName] = useState("");
const [spotNameChangeSuccess, setSpotNameChangeSuccess] = useState(false);
const [isModalOpen, setIsModalOpen] = useState(false);
```